internal/services/dataprotection/data_protection_backup_policy_mysql_flexible_server_resource.go

```diff
@@ -0,0 +1,697 @@
+	VaultId                      string                                                `tfschema:"vault_id"`
+	BackupRepeatingTimeIntervals []string                                              `tfschema:"backup_repeating_time_intervals"`
+	DefaultRetentionRule         []BackupPolicyMySQLFlexibleServerDefaultRetentionRule `tfschema:"default_retention_rule"`
+	RetentionRules               []BackupPolicyMySQLFlexibleServerRetentionRule        `tfschema:"retention_rule"`
+	TimeZone                     string                                                `tfschema:"time_zone"`
+}
+
+type BackupPolicyMySQLFlexibleServerDefaultRetentionRule struct {
+	LifeCycle []BackupPolicyMySQLFlexibleServerLifeCycle `tfschema:"life_cycle"`
+}
+
+type BackupPolicyMySQLFlexibleServerLifeCycle struct {
+	DataStoreType string `tfschema:"data_store_type"`
+	Duration      string `tfschema:"duration"`
+}
+
+type BackupPolicyMySQLFlexibleServerRetentionRule struct {
+	Name      string                                     `tfschema:"name"`
+	Criteria  []BackupPolicyMySQLFlexibleServerCriteria  `tfschema:"criteria"`
+	LifeCycle []BackupPolicyMySQLFlexibleServerLifeCycle `tfschema:"life_cycle"`
+	Priority  int64                                      `tfschema:"priority"`
+}
+
+type BackupPolicyMySQLFlexibleServerCriteria struct {
+	AbsoluteCriteria     string   `tfschema:"absolute_criteria"`
+	DaysOfWeek           []string `tfschema:"days_of_week"`
+	MonthsOfYear         []string `tfschema:"months_of_year"`
+	ScheduledBackupTimes []string `tfschema:"scheduled_backup_times"`
+	WeeksOfMonth         []string `tfschema:"weeks_of_month"`
+}
+
+type DataProtectionBackupPolicyMySQLFlexibleServerResource struct{}
+
+var _ sdk.Resource = DataProtectionBackupPolicyMySQLFlexibleServerResource{}
+
+func (r DataProtectionBackupPolicyMySQLFlexibleServerResource) ResourceType() string {
+	return "azurerm_data_protection_backup_policy_mysql_flexible_server"
+}
+
+func (r DataProtectionBackupPolicyMySQLFlexibleServerResource) ModelObject() interface{} {
+	return &BackupPolicyMySQLFlexibleServerModel{}
+}
+
+func (r DataProtectionBackupPolicyMySQLFlexibleServerResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return backuppolicies.ValidateBackupPolicyID
+}
+
+func (r DataProtectionBackupPolicyMySQLFlexibleServerResource) Arguments() map[string]*pluginsdk.Schema {
+	arguments := map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.BackupPolicyMySQLFlexibleServerName,
+		},
+
+		"vault_id": commonschema.ResourceIDReferenceRequiredForceNew(pointer.To(backuppolicies.BackupVaultId{})),
+
+		"backup_repeating_time_intervals": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			ForceNew: true,
+			MinItems: 1,
+			Elem: &pluginsdk.Schema{
+				Type:         pluginsdk.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
this should be validated as an ISO duration?

Reply:
It's not a standard ISO duration. This is the sample "R/2021-05-23T02:30:00+00:00/P1W". All existing TF resources defined the "backup_repeating_time_intervals" property aren't added validation.

Reply:
the documentation says `ISO 8601 repeating time format` - that is a variation of the standard duration format no?

please create a validation function like `ISO8601Duration` and add it to the helpers/time.go function and make use of it here/add to the other data protection resources

Reply:
The validation is added.